api/swagger.yml

```diff
@@ -1,507 +1,795 @@
-swagger: "2.0"
+openapi: "3.0.0"
 
-consumes:
-  - application/json
 info:
   description: lakeFS HTTP API
   title: lakeFS API
   version: 0.1.0
-produces:
-  - application/json
-schemes:
-  - http
-  - https
-basePath: /api/v1
-
-
-securityDefinitions:
-  basic_auth:
-    type: basic
-  jwt_token:
-    type: apiKey
-    in: header
-    name: X-JWT-Authorization
+
+
+servers:
+  - url: '/api/v1'
+    description: lakeFS server endpoint
+    variables:
+      protocol:
+        enum:
+          - http
+          - https
+        default: http
```

Comment:
I don't understand, when is this used?  (If I did understand, I would want to avoid `http` in favour of `https`, to have the client encrypt by default.)

Reading [server URL format](https://swagger.io/docs/specification/api-host-and-base-path/#url-format), I would expect these variables to have no effect, as they do not appear in the `url`.  Reading further, I would expect
```yaml
servers:
  - url: http://localhost:8000/api/v1
```

to give a single useful default.  Or perhaps

```yaml
    servers:
      - url: http://localhost:8000/api/v1
      - url: '{protocol}://{endpoint}/api/v1'
        variables:
          protocol:
            enum:
              - http
              - https
            default: https
          endpoint:
            default: localhost:8000
```

Reply:
Updated the swagger to include just our base URL and removed the schema information.
Why? - the `kin-openapi` package uses middleware to validate the swagger to match the request, the previous. It tried to verify the full URL match for each request and the endpoint value.
Leaving just the patch match for each request just the api/v1 part.